public/app/plugins/datasource/cloudwatch/components/QueryHeader.test.tsx

```diff
@@ -117,4 +117,93 @@ describe('QueryHeader', () => {
       expect(datasource.resources.isMonitoringAccount).not.toHaveBeenCalledWith();
     });
   });
+
+  describe('when changing query mode', () => {
+    const { datasource } = setupMockedDataSource();
+    it('should set default log query when switching to log mode', async () => {
+      const onChange = jest.fn();
+      datasource.resources.isMonitoringAccount = jest.fn().mockResolvedValue(false);
+      render(
+        <QueryHeader
+          datasource={datasource}
+          query={{ ...validMetricSearchBuilderQuery, expression: 'foo' }}
+          onChange={onChange}
+          onRunQuery={jest.fn()}
+          dataIsStale={false}
+        />
+      );
+      await waitFor(() => expect(screen.queryByText('CloudWatch Metrics')).toBeInTheDocument());
+      await act(async () => {
+        await selectEvent.select(screen.getByLabelText('Query mode'), 'CloudWatch Logs', { container: document.body });
```

Comment:
I think changing `screen.getByLabelText` to `await screen.findByLabelText`, can let you remove the `await act(async () => {` on line 136.

Reply:
done